webapi/src/test/scala/org/knora/webapi/routing/admin/ProjectsRouteZSpec.scala

```diff
@@ -0,0 +1,135 @@
+
+  private val expectMessageToProjectResponderIri: ULayer[ActorToZioBridge] =
+    ActorToZioBridgeMock.AskAppActor
+      .of[ProjectGetResponseADM]
+      .apply(Assertion.equalTo(expectedRequestSuccessIri), Expectation.value(expectedResponseSuccess))
+      .toLayer
+
+  private val expectMessageToProjectResponderShortname: ULayer[ActorToZioBridge] =
+    ActorToZioBridgeMock.AskAppActor
+      .of[ProjectGetResponseADM]
+      .apply(Assertion.equalTo(expectedRequestSuccessShortname), Expectation.value(expectedResponseSuccess))
+      .toLayer
+
+  private val expectMessageToProjectResponderShortcode: ULayer[ActorToZioBridge] =
+    ActorToZioBridgeMock.AskAppActor
+      .of[ProjectGetResponseADM]
+      .apply(Assertion.equalTo(expectedRequestSuccessShortcode), Expectation.value(expectedResponseSuccess))
+      .toLayer
+
+  private val expectNoInteractionWithProjectsResponderADM = ActorToZioBridgeMock.empty
+
+  val spec: Spec[Any, Serializable] =
+    suite("ProjectsRouteZSpec")(
+      test("given valid project iri should respond with success") {
+        val urlWithValidIri = URL.empty.setPath(basePathIri / validIriUrlEncoded)
+        for {
+          route  <- systemUnderTest
+          actual <- route.apply(Request(url = urlWithValidIri)).flatMap(_.body.asString)
+        } yield assertTrue(actual == expectedResponseSuccess.toJsValue.toString())
+      }.provide(commonLayers, expectMessageToProjectResponderIri),
+      test("given valid project shortname should respond with success") {
+        val urlWithValidShortname = URL.empty.setPath(basePathShortname / validShortname)
+        for {
+          route  <- systemUnderTest
+          actual <- route.apply(Request(url = urlWithValidShortname)).flatMap(_.body.asString)
+        } yield assertTrue(actual == expectedResponseSuccess.toJsValue.toString())
+      }.provide(commonLayers, expectMessageToProjectResponderShortname),
+      test("given valid project shortcode should respond with success") {
+        val urlWithValidShortcode = URL.empty.setPath(basePathShortcode / validShortcode)
+        for {
+          route  <- systemUnderTest
+          actual <- route.apply(Request(url = urlWithValidShortcode)).flatMap(_.body.asString)
+        } yield assertTrue(actual == expectedResponseSuccess.toJsValue.toString())
+      }.provide(commonLayers, expectMessageToProjectResponderShortcode),
+      test("given invalid project iri should respond with bad request") {
+        val urlWithInvalidIri = URL.empty.setPath(basePathIri / "invalid")
+        for {
+          route  <- systemUnderTest
+          actual <- route.apply(Request(url = urlWithInvalidIri)).map(_.status)
+        } yield assertTrue(actual == Status.BadRequest)
+      }.provide(commonLayers, expectNoInteractionWithProjectsResponderADM),
+      test("given invalid project shortname should respond with bad request") {
+        val urlWithInvalidShortname = URL.empty.setPath(basePathShortname / "123")
```

Comment:
I am not sure if it would make sense to group tests for a particular path in a common suite.
The test for `basePathShortname` are for example spread out. Usually I would assume I will be on a particular endpoint and would like to have all tests for that in one place.

Later, we could even support grouping different http method into their own `suite`s like:
```
suite("…RouteSpec")(
  suite(s"$somePath")(
    suite("GET")(
      test("given valid…"){…},
      test("given invalid…"){…}
    ), 
    suite("POST")(…)    
  ),
  suite(s"$anotherPath")(…)
)
```

wdyt?

Reply:
I like this idea to sort / distinguish suites based on methods and paths.

Reply:
sounds good, I will restructure it